File: CS/DZ 14.09.22 (Static, Polymorphism)/CS_Polymorphism/CS_Polymorphism/Circle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CS_Polymorphism
{
    class Circle : Figure
    {
        protected int r;

        public Circle(int x, int y, int r) : base(x, y)
        {
            this.r = r;
        }

        public override void Print()
        {
            // вызов метода Print базового класса
            // base.Print();

            Console.WriteLine($"Circle. x = {x}, y = {y}, r = {r}");
        }

        public override double GetS()
        {
            return 3.14 * r*r;
        }
    }
}

File: C++/ДЗ 08.06.22 (полиморфизм)/CPP_Polymorphism/CPP_Polymorphism/Circle.cpp
#include "Circle.h"
#include <iostream>

using namespace std;

Circle::Circle(int x, int y, int r) : Figure(x, y)
{
	cout << "Circle constr" << endl;
	R = r;
}

Circle::~Circle()
{
	cout << "Circle destr" << endl;
}

void Circle::Print()
{
	//Figure::Print();
	cout << "Circle. X = " << X << ", Y = " << Y << ", R = " << R << endl << endl;
}

double Circle::GetS()
{
	return 3.14*R*R;
}